pytorch_lightning/trainer/trainer.py

```diff
@@ -2802,10 +2802,12 @@ def configure_optimizers(self):
 
 @contextmanager
 def _evaluation_context() -> Generator:
-    # inference mode is not supported with gloo backend (#9431)
+    # inference mode is not supported with gloo backend (#9431) and hpu backend
     context_manager_class = (
         torch.inference_mode
-        if _TORCH_GREATER_EQUAL_1_9 and not (dist.is_initialized() and dist.get_backend() == "gloo")
+        if _TORCH_GREATER_EQUAL_1_9
+        and not (dist.is_initialized() and dist.get_backend() == "gloo")
+        and not _HPU_AVAILABLE
```

Comment:
I'd rather suggest conditioning it on if HPU is used or not for the current run?